starboard/nplb/posix_compliance/posix_epoll_test.cc

```diff
@@ -0,0 +1,243 @@
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0)
+      << strerror(errno);
+
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0)
+      << strerror(errno);
+}
+
+TEST_F(PosixEpollTest, SunnyDayMany) {
+  const int kMany = kSbFileMaxOpen;
+  std::vector<int> sockets(kMany, 0);
+
+  for (int i = 0; i < kMany; ++i) {
+    sockets[i] = ::socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+    EXPECT_GE(sockets[i], 0);
+    EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_ADD, sockets[i], &epev), 0)
+        << strerror(errno);
+  }
+
+  for (int i = 0; i < kMany; ++i) {
+    EXPECT_EQ(close(sockets[i]), 0);
+  }
+}
+
+TEST_F(PosixEpollTest, RainyDayRemoveFromEmptyFails) {
+  // First remove should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0);
+}
+
+TEST_F(PosixEpollTest, RainyDayDoubleAddFails) {
+  // First add should succeed.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0)
+      << strerror(errno);
+
+  // Second add should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0);
+
+  // Remove should succeed, to clear epoll queue for next test.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0)
+      << strerror(errno);
+}
+
+TEST_F(PosixEpollTest, RainyDayDoubleRemoveFails) {
+  // First add should succeed.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0)
+      << strerror(errno);
+
+  // Remove should succeed.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0)
+      << strerror(errno);
+
+  // Remove after remove should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0);
+}
+
+TEST_F(PosixEpollTest, RainyDayAddRemoveRepeated) {
+  // First remove should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0);
+
+  // First add should succeed.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0)
+      << strerror(errno);
+
+  // Second add should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0);
+
+  // Remove should succeed.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0)
+      << strerror(errno);
+
+  // Remove after remove should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0);
+
+  // Add after remove should succeed.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0)
+      << strerror(errno);
+
+  // Second add after remove should fail.
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, &epev), 0);
+
+  // Remove should succeed, to clear epoll queue for next test.
+  EXPECT_EQ(epoll_ctl(epfd, EPOLL_CTL_DEL, socket_fd, &epev), 0)
+      << strerror(errno);
+}
+
+TEST_F(PosixEpollTest, RainyDayInvalidSocket) {
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_ADD, -1, &epev), 0);
+}
+
+TEST_F(PosixEpollTest, RainyDayNoInterest) {
+  EXPECT_LT(epoll_ctl(epfd, EPOLL_CTL_ADD, socket_fd, nullptr), 0);
+}
+
+TEST_F(PosixEpollTest, SunnyDayWaitTimeout) {
+  struct epoll_event ev, events[kMaxEvents];
```

Comment:
In C90, you had to define variables at the top of the scope. Since this is C++, move them as close as possible to their use site, e.g. for `ev`, to l.183. Please do this throughout.


And use designated initializers if possible (https://en.cppreference.com/w/cpp/language/aggregate_initialization#Designated_initializers).